### PR TITLE
Improve EGL offscreen support

### PR DESCRIPTION
**Pull request by banesullivan**
Some improvements to better handle OSMesa/EGL offscreen rendering builds of VTK

There may be unintended side effects of these changes

### CODE REVIEW

File: tests/plotting/jupyter/test_trame.py [hunk 1]
```diff
@@ -53,7 +53,11 @@ def test_trame():
     actor = pl.add_mesh(pv.Cone())
     widget = pl.show(return_viewer=True)
     assert isinstance(widget, Widget)
-    assert 'http://' in widget.src
+
+    if pv.global_theme.trame.server_proxy_enabled:
+        assert pv.global_theme.trame.server_proxy_prefix in widget.src
+    else:
+        assert 'http://' in widget.src
```
Comment (adeak) [inline tests/plotting/jupyter/test_trame.py:60]: Intentionally snuck in this PR, or misplaced commit?

Comment (banesullivan) [reply tests/plotting/jupyter/test_trame.py:60]: Intentionally snuck in as the changes here fix other failures in the same testing module

File: pyvista/plotting/plotter.py [hunk 2]
```diff
@@ -296,7 +296,7 @@ def __init__(
         # Keep track of the scale
 
         # track if render window has ever been rendered
-        self._rendered = False
+        self._rendered = uses_egl()
```
Comment (banesullivan) [inline pyvista/plotting/plotter.py:299]: I'm very skeptical of this change.

Comment (banesullivan) [reply pyvista/plotting/plotter.py:299]: Reverting this and merging other changes

Comment (banesullivan) [inline pyvista/plotting/plotter.py:299]:
```suggestion
        self._rendered = False
```

**Pull request merged**